hw/gpio/esp32_gpio.c

```diff
@@ -19,26 +19,197 @@
 #include "hw/qdev-properties.h"
 #include "hw/gpio/esp32_gpio.h"
 
+#include "qapi/qmp/qdict.h"
+#include "qapi/qmp/qjson.h"
+#include "qapi/qmp/qnum.h"
 
+// #define ENABLE_DEBUG
+
+#ifdef ENABLE_DEBUG
+#define D(x) x
+#else
+#define D(x)
```

Comment:
Nitpicking: `#define D(x) do {} while(0)` is a little bit less error-prone. I mean in current solution we can write something like that:
```{.c}
D(int a = 0) // lack of semicolon
D(a++) // lack of semicolon
```
and it's be perfectly fine as long as debug is disabled, but will fail during compilation with debug enabled..

The `do-while` approach fails either way